manifest.go

```diff
@@ -171,10 +196,11 @@ func readManifest(r io.Reader) (*Manifest, []error, error) {
 
 func fromRawManifest(raw rawManifest) (*Manifest, error) {
 	m := &Manifest{
-		Constraints: make(gps.ProjectConstraints, len(raw.Constraints)),
-		Ovr:         make(gps.ProjectConstraints, len(raw.Overrides)),
-		Ignored:     raw.Ignored,
-		Required:    raw.Required,
+		Constraints:  make(gps.ProjectConstraints, len(raw.Constraints)),
+		Ovr:          make(gps.ProjectConstraints, len(raw.Overrides)),
+		Ignored:      raw.Ignored,
+		Required:     raw.Required,
+		PruneOptions: gps.PruneNestedVendorDirs,
```

Comment:
ooooooh, knowing that gps may want to use it differently than dep makes this make a whole lot more sense! 

I don't know what's the best way to make the change, you are much closer to this than I am. But at a high level, having a way for dep to make sure that optional is always passed in properly seems useful.